### PR TITLE
Openssl uefi

### PR DESCRIPTION
**Pull request by qloong**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by richsalz** [comment]
These are bugfixes for a supported build config, so 1.1.0 as well as master.
But @qloong we'll need you to sign an individual CLA as well.  And if you're not LuGong you need to get Dan Zimmerman to add you to the Intel corporate CLA as well.

**Comment by richsalz** [comment]
dang button mistake!

File: ssl/statem/statem_lib.c [hunk 1]
```diff
@@ -279,9 +279,10 @@ MSG_PROCESS_RETURN tls_process_cert_verify(SSL *s, PACKET *pkt)
     const unsigned char *data;
 #ifndef OPENSSL_NO_GOST
     unsigned char *gost_data = NULL;
+    int pktype;
```
Comment (levitte) [inline ssl/statem/statem_lib.c]: Remove this line.

File: ssl/statem/statem_lib.c [hunk 2]
```diff
@@ -384,6 +384,7 @@ MSG_PROCESS_RETURN tls_process_cert_verify(SSL *s, PACKET *pkt)
     }
 #ifndef OPENSSL_NO_GOST
     {
+        pktype = EVP_PKEY_id(pkey);
```
Comment (levitte) [inline ssl/statem/statem_lib.c]: Change this line to `int pktype = EVP_PKEY_id(pkey);`

Comment (levitte) [inline ssl/statem/statem_lib.c]: It's possible that `gost_data` can be moved here in a similar manner

Comment (qloong) [inline ssl/statem/statem_lib.c]: It's OK for me to move them to the position of first usage.
Do we have any rule against the variable declaration: declare all variables in one location or declare them when we first use them?

Comment (richsalz) [inline ssl/statem/statem_lib.c]: we don't have any strong rules.  we try to move things to an inner block if possible, but we don't create blocks *just* to limit variable scope.

**Comment by qloong** [comment]
Done on ICLA & CCLA.

File: ssl/statem/statem_lib.c [hunk 3]
```diff
@@ -277,11 +277,8 @@ MSG_PROCESS_RETURN tls_process_cert_verify(SSL *s, PACKET *pkt)
 {
     EVP_PKEY *pkey = NULL;
     const unsigned char *data;
-#ifndef OPENSSL_NO_GOST
-    unsigned char *gost_data = NULL;
-#endif
```
Comment (levitte) [inline ssl/statem/statem_lib.c]: I'm sorry, I misled you regarding this one. There's now an `OPENSSL_free` further down that's missing its variable...

Comment (qloong) [inline ssl/statem/statem_lib.c]: Exactly. I also didn't notice this issue. So moving the gost_data declaration to the original position.

File: ssl/statem/statem_lib.c [hunk 4]
```diff
@@ -278,10 +278,10 @@ MSG_PROCESS_RETURN tls_process_cert_verify(SSL *s, PACKET *pkt)
     EVP_PKEY *pkey = NULL;
     const unsigned char *data;
 #ifndef OPENSSL_NO_GOST
-    unsigned char *gost_data = NULL;
+    unsigned char *gost_data = NULL;
```
Comment (levitte) [inline ssl/statem/statem_lib.c]: Hmm, just noticed Travis complains here. Seems like it's a hard space, would you mind just rewriting those spaces?

Comment (qloong) [inline ssl/statem/statem_lib.c]: Rewriting spaces. (So silly mistake. Not sure if it's caused by my online-editing.)

**Comment by levitte** [comment]
@qloong, there's one detail missing, as explained in the mail you replied to: you need to get Dan Zimmerman to add you to the Intel corporate CLA as well.

**Comment by mattcaswell** [comment]
CLA now ok. Ping @levitte

**Comment by levitte** [comment]
Merged into master: cff55b90e95e1fa6c90154f93f12363e761d88c7 and dc8da7b1b8ba276f179ea1232c9df764dcd38775

1.1.0 needs a separate PR, the `statem_lib.c` change doesn't cherry-pick cleanly, sorry.

**Pull request closed without merging**